test/unit/utils/api-request.spec.ts

```diff
@@ -132,12 +134,305 @@ function mockRequestWithHttpError(
  *
  * @return {Object} A nock response object.
  */
-function mockRequestWithError(err: Error) {
+function mockRequestWithError(err: any) {
   return nock('https://' + mockHost)
     .get(mockPath)
     .replyWithError(err);
 }
 
+describe('HttpClient', () => {
+  let mockedRequests: nock.Scope[] = [];
+
+  afterEach(() => {
+    _.forEach(mockedRequests, (mockedRequest) => mockedRequest.done());
+    mockedRequests = [];
+  });
+
+  it('should be fulfilled for a 2xx response from an HTTP endpoint', () => {
+    const respData = {foo: 'bar'};
+    const scope = nock('http://' + mockHost)
+      .get(mockPath)
+      .reply(200, respData, {
+        'content-type': 'application/json',
+      });
+    mockedRequests.push(scope);
+    const client = new HttpClient();
+    return client.send({
+      method: 'GET',
+      url: `http://${mockHost}${mockPath}`,
+    }).then((resp) => {
+      expect(resp.status).to.equal(200);
+      expect(resp.headers['content-type']).to.equal('application/json');
+      expect(resp.text).to.equal(JSON.stringify(respData));
+      expect(resp.data).to.deep.equal(respData);
+    });
+  });
+
+  it('should be fulfilled for a 2xx response with a json payload', () => {
```

Comment:
This looks exactly like the test above it.

Reply:
Removed